playbooks/roles/pipeline/tasks/setup_pipelines.yml

```diff
@@ -29,14 +31,15 @@
   shell: "{{ oc_bin }} get buildconfig | grep 'JenkinsPipeline' | awk '{print $1}' | egrep '^{{ item.path.split('/')[-2] }}$'"
   register: "pipeline_name_check"
   with_items: "{{ all_pipelines }}"
+  when: pipelines is defined
   ignore_errors: yes
 
 - name: "Replace pipeline buildconfigs"
-  shell: "{{ oc_bin }} replace -f {{ item.item.path }}"
+  command: "{{ oc_bin }} replace -f {{ item.item.path }}"
   with_items:  "{{ pipeline_name_check.results }}"
   when: "{{ item.rc == 0 }}"
 
 - name: "Create pipeline buildconfigs"
-  shell: "{{ oc_bin }} create -f {{ item.item.path }}"
+  command: "{{ oc_bin }} create -f {{ item.item.path }}"
```

Comment:
I rather use shell in this case since command may have issues in that it is just a command execution

Reply:
Same the above comment. more info here: https://github.com/willthames/ansible-lint/blob/master/lib/ansiblelint/rules/UseCommandInsteadOfShellRule.py

Reply:
Again have you tested it with command vs shell in the setup?